lib/react.js

```diff
@@ -230,6 +230,8 @@ declare module react {
   declare var Component: typeof React$Component;
   declare var PureComponent: typeof React$PureComponent;
   declare var Element: typeof React$Element;
+  
+  declare type Node = string | number | React$Element<any> | Array<Node>;
```

Comment:
Actually I can probably just copy all those type in here, with the exception of some of those Fiber types:

```js
export type Text = string | number;
export type Empty = null | void | boolean;

export type Fragment = Empty | Iterable<Node>;

export type Node =
  | React$Element<any>
  | Text
  | Fragment;

export type NodeList = Empty | Node;
```